src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -26,20 +26,60 @@
 //------------------------------------------------------------------------------
 
 using System;
-using System.Collections.Concurrent;
 using System.Globalization;
 using Microsoft.IdentityModel.Logging;
+#if NET461 || NET472 || NETSTANDARD2_0 
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+#endif
 
 namespace Microsoft.IdentityModel.Tokens
 {
     /// <summary>
     /// Defines a cache for crypto providers.
     /// Current support is limited to <see cref="SignatureProvider"/> only.
     /// </summary>
-    public class InMemoryCryptoProviderCache : CryptoProviderCache
+#if NET461 || NET472 || NETSTANDARD2_0 
+    public class InMemoryCryptoProviderCache : CryptoProviderCache, IDisposable
+#elif NET45
+    public class InMemoryCryptoProviderCache: CryptoProviderCache
+#endif
     {
-        private ConcurrentDictionary<string, SignatureProvider> _signingSignatureProviders = new ConcurrentDictionary<string, SignatureProvider>();
-        private ConcurrentDictionary<string, SignatureProvider> _verifyingSignatureProviders = new ConcurrentDictionary<string, SignatureProvider>();
+        internal CryptoProviderCacheOptions _cryptoProviderCacheOptions;
+#if NET461 || NET472 || NETSTANDARD2_0 
+        private readonly MemoryCache _signingSignatureProviders;
+        private readonly MemoryCache _verifyingSignatureProviders;
+        private bool _disposed = false;
+#elif NET45
+        private readonly MaxCapacityRefreshCache<string, SignatureProvider> _signingSignatureProviders;
+        private readonly MaxCapacityRefreshCache<string, SignatureProvider> _verifyingSignatureProviders;
+#endif
+
+        /// <summary>
+        /// Creates a new instance of <see cref="InMemoryCryptoProviderCache"/> using the default <see cref="CryptoProviderCacheOptions"/>.
+        /// </summary>
+        public InMemoryCryptoProviderCache() : this(new CryptoProviderCacheOptions())
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="InMemoryCryptoProviderCache"/> using the specified <paramref name="cryptoProviderCacheOptions"/>.
+        /// </summary>
+        /// <param name="cryptoProviderCacheOptions">The options which can be used to configure the crypto provider cache.</param>
```

Comment:
suggest: 'The options used to configure <see cref="InMemoryCryptoProviderCache" />.
don't see a reason to include "which can be used to".

Reply:
That's a really good suggestion; the previous comment was far more verbose than it needed to be.